.github/workflows/start-release.yml

```diff
@@ -161,14 +161,32 @@ jobs:
         run: |
           git config user.name "${{github.event.sender.login}}"
           git config user.email noreply@github.com
-      - name: Patch CHANGELOG.md on ${{env.main_branch}} branch
+      - name: Patch CHANGELOG.md on ${{inputs.ref}}
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
           set -u
+          if grep "^## \[${{inputs.version}}\]$" CHANGELOG.md; then
+            echo "\`CHANGELOG.md@${{inputs.ref}}\` has got already the \`[${{inputs.version}}]\` section." >> $GITHUB_STEP_SUMMARY
+            exit 0
+          fi
+
           CHANGELOG_BRANCH="automation/changelog-${{inputs.version}}"
+          TITLE="Advance \`CHANGELOG.md\` to the next release"
+
           if git ls-remote --quiet --exit-code origin "$CHANGELOG_BRANCH"; then
-            echo "Branch \`$CHANGELOG_BRANCH\` already exists."
+            echo "Branch \`$CHANGELOG_BRANCH\` already exists." >> $GITHUB_STEP_SUMMARY
+
+            EXISTING_PR=($(gh pr list --head "$CHANGELOG_BRANCH" --search "$TITLE in:title" --state open --json number,url --jq ".[]|.number,.url"))
+            if [ "${#EXISTING_PR[@]}" -eq 2 ]; then
+              echo ":arrow_right: There is already an open [PR ${EXISTING_PR[0]}](${EXISTING_PR[1]})," \
+                "which needs to be merged, ensuring all lines added to \`CHANGELOG.md\` after the release branch are listed" \
+                "above the \`[${{inputs.version}}]\` section." >> $GITHUB_STEP_SUMMARY
+            else
+              echo ":arrow_right: There is no open PR from this branch. Please open a PR to the ${{env.main_branch}} if" \
+                "\`[CHANGELOG.md](${{github.server_url}}/${{github.repository}}/blob/${{env.main_branch}}/CHANGELOG.md)\`" \
+                "still needs to be updated." >> $GITHUB_STEP_SUMMARY
```

Comment:
change: `to the ${{env.main_branch}} if` to `to the ${{env.main_branch}} branch if`

Reply:
The `automation/changelog-x.x.x` branch is only created by this script, which should also create a PR.

If there is no open PR, it may mean two things:

1. either the PR has been merged;
2. or there were no PR.
 
In both cases it also means that the script is being run for the second time and something went wrong previously. So I guessed in this case we may let the engineer to fix the state, because otherwise we may mess up the changelog more than necessary.

Reply:
I see it like this:

1. Check branch
  - exists? do nothing
  - not exists? create it
 2. Check PR
  - exists? do nothing
  - not exists? open it (we are not merging it, just opening, so no harm is done)

Reply:
additionally we could check whether the pull request has been merged but the branch is not deleted. In this case we could skip opening another PR

Reply:
Don't you mind improving that in the extracted script in the following PRs?

Reply:
Yes, this is fine